### ISSUE TITLE
pam_unix fails to check if account is locked

### ISSUE THREAD

**Issue by profwpollock** [linux-pam/linux-pam#50]
pam_unix should check if an account has been locked (pam_unix_acct.c, in pam_sm_acct_mgmt function).
An ancient assumption seems to be that a locked account has a mangled password hash and thus will fail authentication attempts, so no "locked account" status needed to be checked.  However, with SSH keys, biometrics, etc., authentication will sometimes (perhaps often) bypass checking the password hash.  A simple check for the password hash starting with "!" or "*" would be sufficient.

**Comment by t8m** [comment]
No, we are not going to change the pam_unix behavior in this regard for "ancient assumptions". At most it could be done as non-default option however unless somebody writes a pull request for that it won't happen.

**Comment by profwpollock** [comment]
Would it be better to issue a pull request for a new option on pam_unix, or for a new PAM module to check account status?

Note that if this is not fixed, the man page should be updated to reflect the account status (locked/unlocked) is not checked.  As far as I can tell, the general assumption is that it is checked and people are still attempting to lock accounts using passwd -l or usermod -L.  This of course is a serious security vulnerability.  I would be happy to suggest some wording for this change, if desired.

**Comment by t8m** [comment]
I think adding option on pam_unix should be sufficient of course along with manual page change describing it and describing the default behavior.
